lib/mongo/address.rb

```diff
@@ -202,9 +197,11 @@ def create_resolver(ssl_options)
       error = nil
       ::Socket.getaddrinfo(host, nil, family, ::Socket::SOCK_STREAM).each do |info|
         begin
-          res = FAMILY_MAP[info[4]].new(info[3], port, host)
-          res.socket(connect_timeout, ssl_options).connect!(connect_timeout).close
-          return res
+          specific_address = FAMILY_MAP[info[4]].new(info[3], port, host)
+          socket = specific_address.socket(
+            connect_timeout, ssl_options, connect_timeout: connect_timeout)
```

Comment:
Why doesn't this need to call `connect!` anymore?

Reply:
Sockets are now completely initialized by their constructors.

Reply:
From looking at the diff, I don't see any actual changes in functionality to the constructors. Could you point out to me where the changes were made?

Reply:
https://github.com/mongodb/mongo-ruby-driver/pull/1254/files#diff-ed625ecdabf4aeff5d1a5fc04b958a3aR86 for SSL socket, https://github.com/mongodb/mongo-ruby-driver/pull/1254/files#diff-cc511f0e87683f58982606a59e0f3657R70 for TCP socket, Unix sockets were already completely initialized by their constructor and had a no-op connect! method.

Reply:
Hmm, not sure why I missed them. Thanks!